### PR TITLE
Register kafka serializers and schemaregistry classes only when present

### PR DESCRIPTION
**Pull request by zakkak**
Co-authored-by: @essobedo 

Fixes warnings appearing with https://github.com/quarkusio/quarkus/pull/29886

### CODE REVIEW

File: core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java
Comment (essobedo) [inline core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java:77]:
```suggestion
        return weakClass(true, true, true, className);
```

Comment (zakkak) [reply core/deployment/src/main/java/io/quarkus/deployment/builditem/nativeimage/ReflectiveClassBuildItem.java]: Done

**Review by essobedo — APPROVED** [review]

**Review by alesj — APPROVED** [review]

**Comment by zakkak** [comment]
@gsmet @geoand it looks like I need an approval from a committer as well.

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
Can you rebase on main to get a new CI run please?

**Pull request merged**
